libherokubuildpack/src/buildpack_output/background_printer.rs

```diff
@@ -0,0 +1,195 @@
+//! This module is responsible for the logic involved in the printing to output while
+//! other work is being performed. Such as printing dots while a download is being performed
+use std::io::Write;
+use std::sync::mpsc::{channel, Sender};
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+/// Print dots to the given buffer at the given interval
+///
+/// Returns a struct that allows for manually stopping the timer or will automatically stop
+/// the timer if the guard is dropped. This functionality allows for errors that trigger
+/// an exit of the function to not accidentally have a timer printing in the background
+/// forever.
+#[must_use]
+pub(crate) fn print_interval<W>(
+    mut buffer: W,
+    interval: Duration,
+    start: String,
+    tick: String,
+    end: String,
+) -> PrintGuard<W>
+where
+    W: Write + Send + 'static,
```

Comment:
FWIW the `'static` lifetime requirement really tripped me up. It doesn't mean "lives statically for as long as the program" it means: "Contains only owned values, OR is a staticly declared value." For our use case it means we cannot pass a reference to a `Write` we must always pass the full struct.